astropy/coordinates/representation.py

```diff
@@ -1902,6 +1902,100 @@ def from_cartesian(cls, cart):
         # erfa p2s: P-vector to spherical polar coordinates.
         return cls(*erfa_ufunc.p2s(p), copy=False)
 
+    def transform(self, matrix, *,
+                  cartesian_basis=True, rotation_check=1e-15):
+        r"""
+        Transform the spherical coordinates using a 3x3 matrix.
+
+        This returns a new representation and does not modify the original one.
+        Any differentials attached to this representation will also be
+        transformed.
+
+        Parameters
+        ----------
+        matrix : (3,3) or (2,2) `~numpy.ndarray`
+            A 3x3 transformation matrix, such as a rotation matrix.
+            Can by (2,2) if cartesian_basis is False.
+        cartesian_basis : bool (optional, keyword-only)
+            Whether the matrix is in a spherical or cartesian basis.
+            See Notes for details.
+        rotation_check : float or bool (optional, keyword-only)
```

Comment:
Another option might be to have `None` for "infer from matrix" - I don't know that having the ability to pass a precise threshold is all that useful.

Reply:
I've changed it to that. Now the flag``is_rotation`` can be a bool or None.